### PR TITLE
Small text resizing patch

### PR DESCRIPTION
**Pull request by finben**

### CODE REVIEW

File: Hymns/Display/VerseView.swift [hunk 1]
```diff
@@ -8,15 +8,15 @@ struct VerseLineView: View {
 
     var body: some View {
         HStack(alignment: .top) {
-            Text(viewModel.verseNumber ?? "").font(viewModel.fontSize.font).frame(minWidth: viewModel.fontSize.minWidth)
+            Text(viewModel.verseNumber ?? "").font(viewModel.fontSize.font).frame(minWidth: 12)
```
Comment (finben) [inline Hymns/Display/VerseView.swift:11]: We kind of need a definite size here. This causes the wierd issue especially when you go from big to small text of extra space.

File: Hymns/Display/VerseView.swift [hunk 2]
```diff
@@ -8,15 +8,15 @@ struct VerseLineView: View {
...
             VStack(alignment: .leading) {
                 if transliterate {
                     viewModel.transliteration.map { transliteration in
                         Text(transliteration).font(viewModel.fontSize.font)
                     }
                 }
                 Text(viewModel.verseText).font(viewModel.fontSize.font)
-            }.lineSpacing(5).padding(.bottom, 5)
+                }.fixedSize(horizontal: false, vertical: true).padding(.bottom, 5).lineSpacing(5)
```
Comment (finben) [inline Hymns/Display/VerseView.swift:19]: This deals with the issue of having lyrics do .... and us losing words on songs. This eliminates the issue entirely and just has it wrap to the next line.

File: Hymns/Display/VerseViewModel.swift [hunk 3]
```diff
@@ -22,9 +22,7 @@ class VerseLineViewModel: Hashable, ObservableObject {
         userDefaultsManager
             .fontSizeSubject
             .sink { fontSize in
-                withAnimation(.easeInOut(duration: 0.5)) {
-                    self.fontSize = fontSize
-                }
+                self.fontSize = fontSize
         }.store(in: &disposables)
```
Comment (finben) [inline Hymns/Display/VerseViewModel.swift:26]: I could not find a single animation that results in us not having the ... during transition. To me it's probably better to just forgo the animation all together than have it feed in the odd manner it is now.

**Comment by skywalkerdude** [comment]
This doesn't fix the issue where the text isn't aligned correctly after resizing from large/extra large to normal:
Simulator Screen Shot - iPhone 8 - 2020-06-14 at 17 02 38 vs 
Simulator Screen Shot - iPhone 8 - 2020-06-14 at 17 02 33

File: Hymns/Display/VerseViewModel.swift (hunk 3 above)
Comment (skywalkerdude) [reply Hymns/Display/VerseViewModel.swift:26]: agreed

File: Hymns/Display/VerseView.swift (hunk 2 above)
Comment (skywalkerdude) [reply Hymns/Display/VerseView.swift:19]: Without the animation, the "..." never appears anymore, so I think we can get rid of this

File: Hymns/Display/VerseView.swift (hunk 1 above)
Comment (skywalkerdude) [reply Hymns/Display/VerseView.swift:11]: I still get the extra space: Simulator Screen Shot - iPhone 8 - 2020-06-14 at 17 02 33

File: Hymns/Display/VerseView.swift (hunk 2 above)
Comment (finben) [reply Hymns/Display/VerseView.swift:19]: It happens on animation but it also happens when a lines text does not fit. For instance try an iphone 8 emulation and the song "in this godless age."

**Comment by finben** [comment]
Okay, I got a fix... Just don't ask me how it works... lol. Also, take a look at this thing I found to trim the whitespaces. I feel like its nice one hymns like 498 with janky whitespace coming from hymnal.net. But you may not agree.

File: Hymns/Display/HymnLyricsView.swift [hunk 4]
```diff
@@ -36,7 +36,7 @@ public struct HymnLyricsView: View {
                         ForEach(lyrics, id: \.self) { verseViewModel in
                             VerseView(viewModel: verseViewModel, transliterate: self.$transliterate)
                         }
-                    }.padding()
+                    }.frame(minWidth: 0, maxWidth: .infinity, alignment: .leading).padding()
```
Comment (skywalkerdude) [inline Hymns/Display/HymnLyricsView.swift:39]: I think this is the key line. Everything else, I removed it, and it seemed to still work

File: Hymns/Display/VerseView.swift (hunk 1 above)
Comment (skywalkerdude) [reply Hymns/Display/VerseView.swift:11]: I think we can revert this right?

File: Hymns/Display/VerseView.swift (hunk 2 above)
Comment (skywalkerdude) [reply Hymns/Display/VerseView.swift]: ah, gotcha. you are right!

File: Hymns/Display/VerseView.swift [hunk 5]
```diff
@@ -8,15 +8,17 @@ struct VerseLineView: View {
...
             VStack(alignment: .leading) {
                 if transliterate {
                     viewModel.transliteration.map { transliteration in
-                        Text(transliteration).font(viewModel.fontSize.font)
+                        Text(transliteration.trimmingCharacters(in: .whitespacesAndNewlines)).font(viewModel.fontSize.font)
                     }
                 }
-                Text(viewModel.verseText).font(viewModel.fontSize.font)
-                }.fixedSize(horizontal: false, vertical: true).padding(.bottom, 5).lineSpacing(5)
+                Text(viewModel.verseText.trimmingCharacters(in: .whitespacesAndNewlines)).font(viewModel.fontSize.font)
```
Comment (skywalkerdude) [inline Hymns/Display/VerseView.swift]: I think the whitespace is okay. It makes sense to indent some of the lines (even in the hymnal they're indented), so I would revert this part.

File: Hymns/Display/HymnLyricsView.swift [hunk 6]
```diff
@@ -13,11 +13,11 @@ public struct HymnLyricsView: View {
             guard let lyrics = viewModel.lyrics else {
                 return Text("Lyrics are not available for this song").maxSize().eraseToAnyView()
             }
-
+
```
Comment (skywalkerdude) [inline Hymns/Display/HymnLyricsView.swift]: swiftlint

File: Hymns/Display/HymnLyricsView.swift [hunk 7]
```diff
@@ -13,11 +13,11 @@ public struct HymnLyricsView: View {
             guard let lyrics = viewModel.lyrics else {
                 return Text("Lyrics are not available for this song").maxSize().eraseToAnyView()
             }
-
+            
             guard !lyrics.isEmpty else {
                 return ActivityIndicator().maxSize().eraseToAnyView()
             }
-
+
```
Comment (skywalkerdude) [inline Hymns/Display/HymnLyricsView.swift]: swiftlint

File: Hymns/Display/HymnLyricsView.swift (hunk 4 above)
Comment (finben) [reply Hymns/Display/HymnLyricsView.swift:39]: Yeah, that is the line.

File: Hymns/Display/VerseView.swift (hunk 5 above)
Comment (finben) [reply Hymns/Display/VerseView.swift]: reverted in recent commit

File: Hymns/Display/HymnLyricsView.swift (hunk 6 above)
Comment (finben) [reply Hymns/Display/HymnLyricsView.swift]: addressed

**Pull request merged**